File: q2918/q2918.java
class Solution {
    public long minSum(int[] nums1, int[] nums2) {
        long sum1 = 0; 
        long sum2 = 0; 
        long zero1 = 0; 
        long zero2 = 0;
        for (int num : nums1) {
            sum1 += num;
            zero1 += (num == 0 ? 1 : 0);
        }

        for (int num : nums2) {
            sum2 += num;
            zero2 += (num == 0 ? 1 : 0);
        }

        // If sum1 and sum2 are already equal 
        if (zero1 == 0 && zero2 == 0) return sum1 == sum2 ? sum1 : -1;

        if (zero1 == 0) return sum2 + zero2 <= sum1 ? sum1 : -1;

        if (zero2 == 0) return sum1 + zero1 <= sum2 ? sum2 : -1;

        return Math.max(sum1 + zero1, sum2 + zero2);
    }
}

File: q2918/q2918.go
func minSum(nums1 []int, nums2 []int) int64 {
    var sum1, sum2, zero1, zero2 int64 = 0, 0, 0, 0
    for _, num := range nums1 {
        sum1 += int64(num)
        if num == 0 {
            zero1++
        }
    }

    for _, num := range nums2 {
        sum2 += int64(num)
        if num == 0 {
            zero2++
        }
    }

    if zero1 == 0 && zero2 == 0 {
        if sum1 == sum2 {
            return sum1
        } 
        
        return -1
    } 
    
    if (zero1 == 0) {
        if (sum2 + zero2 <= sum1) {
            return sum1
        } 

        return -1
    }

    if (zero2 == 0) {
        if (sum1 + zero1 <= sum2) {
            return sum2
        }
        
        return -1
    }

    
    return max(sum1 + zero1, sum2 + zero2)
}

func max(a, b int64) int64 {
    if a < b {
        return b
    }
    return a
}
